src/bun.js/api/server.zig

```diff
@@ -1390,6 +1397,26 @@ pub const ServerConfig = struct {
             }
             if (global.hasException()) return error.JSError;
 
+            if (arg.getTruthy(global, "app")) |bake_args_js| {
+                if (!bun.FeatureFlags.bake) {
+                    global.throwInvalidArguments("To use the experimental \"app\" option, upgrade to the canary build of bun via \"bun upgrade --canary\"", .{});
+                    return;
```

Comment:
these 4 returns should be `error.JSError`

Reply:
or `global.throwInvalidArguments2`

Reply:
i like to embrace the most robust option

![image](https://github.com/user-attachments/assets/6db918b5-7d55-4e02-adc8-b91a21f6058f)